packages/nimble-components/src/rich-text/editor/tests/rich-text-editor.spec.ts

```diff
@@ -1052,6 +1052,395 @@ describe('RichTextEditor', () => {
+                        url: 'https://nimble.ni.dev',
+                        textContent: 'Anchor between https://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Hyperlink(http) within paragraph node',
+                        input: '<p>Anchor between <a href="http://nimble.ni.dev">Nimble</a> text</p>',
+                        url: 'http://nimble.ni.dev',
+                        textContent: 'Anchor between http://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Absolute link(https) as nimble-anchor within paragraph node',
+                        input: '<p>Anchor between <nimble-anchor href="https://nimble.ni.dev"><a href="https://nimble.ni.dev">https://nimble.ni.dev</a></nimble-anchor> text</p>',
+                        url: 'https://nimble.ni.dev',
+                        textContent: 'Anchor between https://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Hyperlink(http) as nimble-anchor within paragraph node',
+                        input: '<p>Anchor between <nimble-anchor href="http://nimble.ni.dev"><a href="http://nimble.ni.dev">Nimble</a></nimble-anchor> text</p>',
+                        url: 'http://nimble.ni.dev',
+                        textContent: 'Anchor between http://nimble.ni.dev text'
+                    }
+                ] as const;
+
+                parameterizeNamedList(
+                    differentProtocolLinks,
+                    (spec, name, value) => {
+                        spec(
+                            `${name} renders as absolute link(href and text content should be same) in editor`,
+                            () => {
+                                const editor = element.shadowRoot?.querySelector(
+                                    '.ProseMirror'
+                                );
+                                pageObject.pasteHTMLToEditor(value.input);
+
+                                expect(pageObject.getEditorTagNames()).toEqual([
+                                    'P',
+                                    'A'
+                                ]);
+                                expect(editor!.textContent).toEqual(
+                                    value.textContent
+                                );
+                                expect(
+                                    pageObject.getEditorLastChildAttribute(
+                                        'href'
+                                    )
+                                ).toBe(value.url);
+                            }
+                        );
+                    }
+                );
+            });
+
+            describe('pasting various valid(https/http) nimble-anchor links should render as absolute HTML anchors', () => {
+                const differentProtocolLinks = [
```

Comment:
Again, I'm confused by the nested links, and I'm skeptical that all these permutations are necessary.

Reply:
Removed these permutations of tests